django/db/models/sql/compiler.py

```diff
@@ -852,11 +856,12 @@ def execute_sql(self, result_type=MULTI):
             cursor, self.connection.features.empty_fetchmany_value,
             self.col_count
         )
-        if not self.connection.features.can_use_chunked_reads:
+        if not self.connection.features.can_use_chunked_reads and not chunked_fetch:
             try:
                 # If we are using non-chunked reads, we return the same data
                 # structure as normally, but ensure it is all read into memory
-                # before going any further.
+                # before going any further. If the user requests chunked_fetch
+                # we will honor that.
                 return list(result)
             finally:
                 # done with the cursor
```

Comment:
Not related to this patch but I wonder if the `cursor.close()` below should also be wrapped?